spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/annotation/SleuthSpanCreatorAspectNegativeTests.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.annotation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanReporter;
+import org.springframework.cloud.sleuth.annotation.SleuthSpanCreatorAspectNegativeTests.TestConfiguration;
+import org.springframework.cloud.sleuth.assertions.ListOfSpans;
+import org.springframework.cloud.sleuth.util.ArrayListSpanAccumulator;
+import org.springframework.cloud.sleuth.util.ExceptionUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.springframework.cloud.sleuth.assertions.SleuthAssertions.then;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = TestConfiguration.class)
+public class SleuthSpanCreatorAspectNegativeTests {
+
+	@Autowired NotAnnotatedTestBeanInterface testBean;
+	@Autowired TestBeanInterface annotatedTestBean;
+	@Autowired ArrayListSpanAccumulator accumulator;
+
+	@Before
+	public void setup() {
+		ExceptionUtils.setFail(true);
+		this.accumulator.clear();
+	}
+
+	@Test
+	public void shouldNotCallAdviceForNotAnnotatedBean() {
+		this.testBean.testMethod();
+
+		then(this.accumulator.getSpans()).isEmpty();
+		then(ExceptionUtils.getLastException()).isNull();
+	}
+
+	@Test
+	public void shouldCallAdviceForAnnotatedBean() throws Throwable {
+		this.annotatedTestBean.testMethod();
+
+		List<Span> spans = new ArrayList<>(this.accumulator.getSpans());
+		then(new ListOfSpans(spans)).hasSize(1).hasASpanWithName("testMethod");
```

Comment:
@Koizumi85 I've also change testing of the implementation vs checking the results. We don't care if some methods got executed. We care that a span with a given name was created.